src/ast/descriptor.ts

```diff
@@ -0,0 +1,8 @@
+import * as jsdoc from '../ast-utils/jsdoc';
+
+export type LiteralValue = string|number|boolean|RegExp;
+
+export interface Descriptor {
```

Comment:
For incrementality we're gonna need to know the path/url of this descriptor. We'll also need to be able to walk to the the entities that depend on this entity (e.g. if this entity is a behavior then we'd need to be able to walk to all the upstream entities that mix it in to refresh them based on its updated state).